### PR TITLE
Alias improvements

### LINKED ISSUE #56: Aliases can be directly re-registred after deletion
If I delete an alias, another user can directly re-register it. I guess, that's not, what we wanted. Also, there are entries in the database left behind without `user_id`, which will probably never be deleted.

On the other hand:
If we allow users to create and delete aliases, but don't free up aliases after deletion,  1 single user could exhaust the complete namespace by just creating and deleting all (short) addresses.

### PR DESCRIPTION
**Pull request by t2d**
Fix #56 and #60

### CODE REVIEW

File: src/Repository/AliasRepository.php [hunk 1]
```diff
@@ -20,6 +20,16 @@ public function findOneBySource($email)
         return $this->findOneBy(array('source' => $email));
     }
 
+    /**
+     * @param $email
+     *
+     * @return array|Alias[]
+     */
+    public function findAllBySource($email)
```
Comment (0x46616c6b) [inline src/Repository/AliasRepository.php]: We should add a second parameter for deleted which is default `false`

File: config/doctrine/Alias.orm.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ App\Entity\Alias:
       nullable: false
     destination:
       type: string
-      nullable: false
+      nullable: true
```
Comment (0x46616c6b) [inline config/doctrine/Alias.orm.yml]: We should not allow the destination to be empty, I guess postfix will use this field. We should ensure the field is filled by the user or we take the email from the user in there.

File: src/Admin/AliasAdmin.php [hunk 3]
```diff
@@ -98,6 +119,10 @@ protected function configureBatchActions($actions)
     public function prePersist($alias)
     {
         if (null == $alias->getDestination()) {
+            if (null == $alias->getUser()) {
+                # set user_id to current user if neither destination nor user_id is given
+                $alias->setUser($this->storage->getToken()->getUser());
```
Comment (0x46616c6b) [inline src/Admin/AliasAdmin.php]: Here you should set the destination with the email from the user.

File: src/Entity/Alias.php [hunk 4]
```diff
@@ -13,7 +13,7 @@
 /**
  * @author louis <louis@systemli.org>
  */
-class Alias implements SoftDeletableInterface
```
Comment (0x46616c6b) [inline src/Entity/Alias.php]: Why? This reverted the complete soft delete functionality.

File: config/doctrine/Alias.orm.yml (hunk 2 above)
Comment (t2d) [reply config/doctrine/Alias.orm.yml]: But it should be empty for privacy reasons after the alias has been deleted.

File: src/Entity/Alias.php (hunk 4 above)
Comment (t2d) [reply src/Entity/Alias.php]: No, it doesn't. The actual functionality is in `DeleteTrait`.

File: src/Admin/AliasAdmin.php (hunk 3 above)
Comment (t2d) [reply src/Admin/AliasAdmin.php]: That's excactly what I'm doing. But if no user is given, I'm getting the user first from the current admin. Afterwards the destination is still updated.

File: src/Entity/Alias.php (hunk 4 above)
Comment (0x46616c6b) [reply src/Entity/Alias.php]: But we check in BaseRepository if the Class is instance of the Interface.

File: src/Admin/AliasAdmin.php (hunk 3 above)
Comment (0x46616c6b) [reply src/Admin/AliasAdmin.php]: Ah okay, its a bit tricky to unterstand.

File: src/Entity/Alias.php (hunk 4 above)
Comment (t2d) [reply src/Entity/Alias.php]: I see. I did this because of #61. Now it works similar to the User Entity.

Comment (t2d) [reply src/Entity/Alias.php]: lets do it in another PR

**Pull request merged**